gobblin-data-management/src/main/java/org/apache/gobblin/util/commit/CreateAndSetDirectoryPermissionCommitStep.java

```diff
@@ -66,9 +68,13 @@ public void execute() throws IOException {
     for (Map.Entry<String, OwnerAndPermission> entry : pathAndPermissions.entrySet()) {
       Path path = new Path(entry.getKey());
       try {
-        log.info("Setting permission {} on path {}", entry.getValue().getFsPermission(), path);
-        fs.setPermission(path, entry.getValue().getFsPermission());
-        // TODO : we can also set owner and group here.
+        if (!fs.exists(path)) {
+          log.info("Creating path {} with permission {}", path, entry.getValue().getFsPermission());
+          fs.mkdirs(path, entry.getValue().getFsPermission());
```

Comment:
removing the todo?

Reply:
It sets the owner and group when you add permissions right https://hadoop.apache.org/docs/r2.6.4/api/org/apache/hadoop/fs/permission/FsPermission.html